Fix inconsistencies in RFC

dplumb94 opened the pull request:
Fixes several small issues with the RFC

Commit:
Update addressing

The addressing referenced here was a pre-2.0 scheme.

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

shannynalayna commented on text/0025-purchase-order.md:
```diff
@@ -551,8 +552,9 @@ message PurchaseOrderPayload {
 }
 
 message CreatePurchaseOrderPayload {
```
Doesn't this message also need an optional `CreateVersionPayload`?

dplumb94 replied:
Ah yeah I was not originally going to make that change here, but I'll add it.

shannynalayna commented on text/0025-purchase-order.md:
```diff
@@ -551,8 +552,9 @@ message PurchaseOrderPayload {
+  uint64 created_at = 3;
 }
 
 message UpdatePurchaseOrderPayload {
```
Would this message also need a `po_uuid`?

dplumb94 replied:
That is just the `uuid` here

shannynalayna replied:
I'm slightly confused where the `uuid` is coming from in this message?

vaporos commented on text/0025-purchase-order.md:
```diff
@@ -638,31 +638,30 @@ Validation Requirements
 with a new revision.
 
 Validation Requirements:
-- The `org_id` must exist in Pike for it be a valid transaction 
```
This looks unrelated to this commit.

dplumb94 replied:
moved to the proper commit

Commit:
Fix typos, capitalization, and spacing errors

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

Commit:
Change `uuid` to `uid`

We don't necessarily know that this IDs will be a "uuid" as
conventionally defined. A more accurate term would be "uid".

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

Commit:
Remove public_key from transaction payload

This should be derived from the transaction signer.

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

Commit:
Add CreateVersionPayload to state

This is detailed in the description but the field did not exist.

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

Commit:
Add `po_uid` field to txn payloads

Without this, we don't know which PO to update or add a version to.

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

Commit:
Add protobuf syntax highlighting

Signed-off-by: Darian Plumb <dplumb@bitwise.io>

shannynalayna commented on text/0025-purchase-order.md:
```diff
@@ -566,8 +566,9 @@ message UpdatePurchaseOrderPayload {
 
 message CreateVersionPayload {
```
In the same vein as this commit, I believe the `UpdatePurchaseOrderPayload` needs a `po_uid` field

shannynalayna commented on text/0025-purchase-order.md:
```diff
@@ -566,8 +566,9 @@ message UpdatePurchaseOrderPayload {
+  PayloadRevision revision = 4;
 }
 
 message UpdateVersionPayload {
```
Also, this payload needs a `po_uid` to access the purchase order in state
